enaml/core/compiler_common.py

```diff
@@ -679,13 +677,25 @@ def gen_operator_binding(cg, node, index, name):
 
     """
     mode = COMPILE_MODE[type(node.value)]
-    global_vars, has_comp = analyse_globals_and_comps(node.value.ast)
-    code = compile(node.value.ast, cg.filename, mode=mode)
+    global_vars, has_defs = analyse_globals_and_func_defs(node.value.ast)
+    # In mode exec, the body of the operator has been wrapped in a function def
+    # after the compilation we extract the function code
+    if mode == 'exec':
+        code = compile(node.value.ast, cg.filename, mode=mode)
+        for op, op_arg in bp.Code.from_code(code).code:
+            if isinstance(op_arg, bp.Code):
+                rewrite_globals_access(op_arg, global_vars)
```

Comment:
Is this duplicating logic from line 697 below?

Reply:
Looks like it is indeed. I will fix it.